addons/binding/org.openhab.binding.hdanywhere/src/main/java/org/openhab/binding/hdanywhere/handler/Mhub4K431Handler.java

```diff
@@ -0,0 +1,169 @@
+                        }.getType();
+                        Map<String, String> map = gson.fromJson(response, type);
+
+                        String inputChannel = map.get("Inputchannel");
+
+                        for (int i = 0; i < numberOfPorts; i++) {
+                            DecimalType decimalType = new DecimalType(String.valueOf(inputChannel.charAt(i)));
+                            updateState(new ChannelUID(getThing().getUID(), Port.get(i + 1).channelID()), decimalType);
+                        }
+                    } else {
+                        updateStatus(ThingStatus.OFFLINE);
+                    }
+                }
+            } catch (Exception e) {
+                logger.warn("An exception occurred while polling the HDanwywhere matrix: '{}'", e.getMessage());
+            }
+        }
+    };
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+
+        if (command instanceof RefreshType) {
+            // Simply schedule a single run of the polling runnable to refresh all channels
+            scheduler.schedule(pollingRunnable, 0, TimeUnit.SECONDS);
+        } else {
+
+            String channelID = channelUID.getId();
+
+            String host = (String) getConfig().get(IP_ADDRESS);
+            int sourcePort = Integer.valueOf(command.toString());
+            int outputPort = Port.get(channelID).toNumber();
+
+            if (sourcePort > numberOfPorts) {
+                // nice try - we can switch to a port that does not physically exist
+                logger.warn("Source port {} goes beyond the physical number of {} ports available on the matrix {}",
+                        new Object[] { sourcePort, numberOfPorts, host });
+            } else if (outputPort > numberOfPorts) {
+                // nice try - we can switch to a port that does not physically exist
+                logger.warn("Output port {} goes beyond the physical number of {} ports available on the matrix {}",
+                        new Object[] { outputPort, numberOfPorts, host });
+            } else {
+
+                String httpMethod = "POST";
+                String url = "http://" + host + "/cgi-bin/MMX32_Keyvalue.cgi";
+
+                String content = "{CMD=";
+                content = content + command.toString() + "B";
+                content = content + String.valueOf(outputPort) + ".";
+
+                InputStream stream = new ByteArrayInputStream(content.getBytes(StandardCharsets.UTF_8));
+
+                Properties httpHeaders = new Properties();
+                httpHeaders.setProperty("Cookie", "logintype-88=01");
+
+                try {
+                    String response = HttpUtil.executeUrl(httpMethod, url, httpHeaders, stream,
+                            "application/x-www-form-urlencoded; charset=UTF-8", timeout);
+                } catch (IOException e) {
+                    logger.error("Communication with device failed: {}", e);
```

Comment:
could you set the ThingStatus to OFFLINE with an according message and reduce the logging to debug?